analytics/views.py

```diff
@@ -1182,6 +1185,17 @@ def support(request: HttpRequest) -> HttpResponse:
         for realm in realms:
             realm.customer = get_customer_by_realm(realm)
 
+            current_plan = get_current_plan_by_realm(realm)
+            if current_plan is not None:
+                new_plan, last_ledger_entry = make_end_of_cycle_updates_if_needed(current_plan, timezone_now())
```

Comment:
Doesn't it have some potential pitfalls to make the act of previewing a realm on /activity/support/ mutate the state of the plan through `make_end_of_cycle_updates_if_needed`? Can't we get the information we need without side-effects?

Reply:
Yeah, I'd really like to avoid any mutation-on-view in this (or really any other) UI.

Reply:
I added `make_end_of_cycle_update` to make sure that the plan details are up to date. Because the cron job that invoice plans (which also calls `make_end_of_cycle_update`)  runs only once every day. 

For example, let's say we have a plan that is supposed to be expired/renewed on September 1 at 10 AM.

And lets assume the last time the cron job run was on September 1 at 6 AM,

If we search for the plan details at 11AM, the info showed would be outdated if we don't call `make_end_of_cycle_update`. That is the reason we have that call in /billing page as well.